Tests/SwiftParserTest/StatementTests.swift

```diff
@@ -957,4 +957,16 @@ final class StatementTests: ParserTestCase {
       ]
     )
   }
+
+  func testForUnsafeStatement() {
+    assertParse(
+      "for try await unsafe x in e { }",
+      experimentalFeatures: [.unsafeExpression]
+    )
+
+    assertParse(
+      "for try await unsafe in e { }",
+      experimentalFeatures: [.unsafeExpression]
+    )
```

Comment:
Could you also add a test for `for unsafe in e { }`?

Reply:
Sure. There's another ambiguity I also need to check for, i.e., `for unsafe: Int in e { }` needs to not parse as the `unsafe` keyword.